core/impl/InstanceManager.ts

```diff
@@ -44,6 +47,20 @@ class InstanceManager {
         wait().thenAction(() => this.stores.delete(s));
     }
 
+    registerModelWithTestId(testId: string, m: HoistModel) {
+        this.modelsByTestId.set(testId, m);
+    }
+
+    unregisterModelWithTestId(testId: string) {
+        this.modelsByTestId.delete(testId);
+    }
+
+    getModelByTestId(testId: string): HoistModel {
+        const model = this.modelsByTestId.get(testId);
+        throwIf(!model, `No model found for component with testId: ${testId}`);
```

Comment:
Would suggest we return null vs throw here.

Reply:
Updated to return `undefined` in that case. Could also fallback on `null` if you think that's better than `undefined`?

Reply:
for what its worth, I almost *always* prefer null to undefined.  Certainly the java programmer in me, but its also a practice in Hoist generallyt